src/Connector.AzureDataLake/Connector/AzureDataLakeConnector.cs

```diff
@@ -214,16 +507,52 @@ public override IReadOnlyCollection<StreamMode> GetSupportedModes()
             return new[] { StreamMode.Sync, StreamMode.EventStream };
         }
 
-        public override async Task RemoveContainer(ExecutionContext executionContext, IReadOnlyStreamModel streamModel)
+        public override Task RemoveContainer(ExecutionContext executionContext, IReadOnlyStreamModel streamModel)
         {
             _logger.LogInformation($"AzureDataLakeConnector.RemoveContainer: entry");
 
             throw new NotImplementedException(nameof(RemoveContainer));
         }
 
-        public virtual async Task<IConnectorConnectionV2> GetAuthenticationDetails(ExecutionContext executionContext, Guid providerDefinitionId)
+        private async Task DeleteCacheTableIfExists(ExecutionContext executionContext, IReadOnlyStreamModel streamModel)
         {
-            return await AuthenticationDetailsHelper.GetAuthenticationDetails(executionContext, providerDefinitionId);
+            var providerDefinitionId = streamModel.ConnectorProviderDefinitionId!.Value;
+            var containerName = streamModel.ContainerName;
+
+            var jobData = await AzureDataLakeConnectorJobData.Create(executionContext, providerDefinitionId, containerName);
+            if (string.IsNullOrWhiteSpace(jobData.StreamCacheConnectionString))
+            {
+                _logger.LogDebug("Skipping deletion of cache table because stream cache connection string is null or whitespace.");
+                return;
+            }
+
+            await using var connection = new SqlConnection(jobData.StreamCacheConnectionString);
+            await connection.OpenAsync();
+            var tableName = GetCacheTableName(streamModel.Id);
+            await DeleteCacheTableIfExists(connection, tableName);
+        }
+
+        private static async Task DeleteCacheTableIfExists(SqlConnection connection, string tableName)
+        {
+            var deleteTableSql = $"""
```

Comment:
Should be done in a transaction so we don't end up in a state where the table is deleted but the history tabale is not